spec/datetimeformat.html

```diff
@@ -514,6 +519,9 @@ <h1>Internal slots</h1>
             <li>month, day</li>
             <li>hour, minute, second</li>
             <li>hour, minute</li>
+            <li>dayPeriod, hour</li>
+            <li>dayPeriod, hour, minute, second</li>
+            <li>dayPeriod, hour, minute</li>
           </ul>
           Each of the records must also have a pattern field, whose value is a String value that contains for each of the date and time format component fields of the record a substring starting with `"{"`, followed by the name of the field, followed by `"}"`. If the record has an hour field, it must also have a pattern12 field, whose value is a String value that, in addition to the substrings of the pattern field, contains a substring `"{ampm}"`.
```

Comment:
This part seems problematic, doesn't it? The current spec requires a `pattern12` field with an `"{ampm}"` substring for any record which has an hour field. But I don't see how this should work with day-period, because then the record must contain day-period and an am/pm marker, which doesn't really make sense, does it?

For example I currently get in SpiderMonkey with the day-period patch enabled:
```js
js> new Intl.DateTimeFormat("en", {hour:"numeric", dayPeriod:"long"}).format()
"1 in the afternoon"
```

But the current spec text kind of implies the output should rather be:
```js
js> new Intl.DateTimeFormat("en", {hour:"numeric", dayPeriod:"long"}).format()
"1 PM in the afternoon"
```

And <https://unicode.org/reports/tr35/tr35-dates.html#availableFormats_appendItems> also seems to disallow using "B" and "a" in a single skeleton:
> Only one field of each type is allowed; that is, "Hh" is not valid.

Reply:
Good catch; I agree we should fix this. Do you have an idea for wording which would capture that same constraint as CLDR skeletons have?